lib/+wbs/@Contacts/Contacts.m

```diff
@@ -192,21 +195,42 @@
 
         function forces = compute_unilateral_linear_contact(obj, J_feet, M, h, torque, JDot_nu_feet, contact_point, generalized_ext_wrench)
             % compute_unilateral_linear_contact returns the pure forces acting on the feet vertices
-
+            
+            persistent firstSolverIter;
+            
+            if isempty(firstSolverIter)
+                firstSolverIter = true;
+            end
+            
             free_acceleration = obj.compute_free_acceleration(M, h, torque, generalized_ext_wrench);
             free_contact_acceleration = obj.compute_free_contact_acceleration(J_feet, free_acceleration, JDot_nu_feet);
             H = J_feet * (M \ J_feet');
-
+            
             if ~issymmetric(H)
                 H = (H + H') / 2; % if non sym
             end
-
+            
             for i = 1:obj.num_vertices * 2
                 obj.Aeq(i, i * 3) = contact_point(i) > 0;
+%                 obj.Aeq(i,i*3-2:i*3) = contact_point(i) > 0;
+            end
+            
+            if obj.useOSQP
+                if firstSolverIter
+                    % Setup workspace and change alpha parameter
+                    obj.osqpProb.setup(sparse(H), free_contact_acceleration, sparse([obj.A;obj.Aeq]), [obj.AxLb;obj.beq], [obj.AxUb;obj.beq], 'alpha', 1);
```

Comment:
Out of curiosity, why are you using sparse ?

Reply:
It was as per the [example](https://osqp.org/docs/examples/setup-and-solve.html) in the tool HowTo. Actually I noticed, after looking into their MATLAB code, that they re-compute the sparse matrix from the inputs we give to `osqpProb.setup`. Anyway, it's better to stick to the interface recommendations.

For info, if we define
```
A = sparse(<some matrix with zeros and non-zero elements>)
```
although `A` stores only the non-zero elements of the matrix, any operation on `A` is seamless, i.e. appears as if `A` was defined normally.

Reply:
Yes I think it speeds up anyway the computation for the solver

Reply:
Yes sure, that's the whole point of the use of sparse matrices.

Reply:
But actually for now, doing it at our level is kind of useless because they redo it in the API methods. Their API is not so well documented regarding these details.